OpenRA.Mods.Common/Traits/World/Selection.cs

```diff
@@ -15,19 +15,26 @@
 using OpenRA.Primitives;
 using OpenRA.Traits;
 
-namespace OpenRA
+namespace OpenRA.Mods.Common.Traits
 {
-	public class Selection
+	public class SelectionInfo : ITraitInfo
+	{
+		public object Create(ActorInitializer init) { return new Selection(this); }
+	}
+
+	public class Selection : ISelection, INotifyCreated, INotifyOwnerChanged, ITick, IGameSaveTraitData
 	{
 		public int Hash { get; private set; }
 		public IEnumerable<Actor> Actors { get { return actors; } }
 
 		readonly HashSet<Actor> actors = new HashSet<Actor>();
-		readonly INotifySelection[] worldNotifySelection;
+		INotifySelection[] worldNotifySelection = { };
```

Comment:
Please change this to just `INotifySelection[] worldNotifySelection;` - we *should* be crashing if something tries to access this before `Created` assigns a value.